### ISSUE TITLE
Komodo hangs while adding blank lines

### ISSUE THREAD

**Issue by pwo311** [Komodo/KomodoEdit#2430]
### Short Summary

While editing Komodo crashes. I can recreate the problem by literally adding lines of generic code or just simple blank lines. 

I can recreate the same problem on completely different files.

### Steps to Reproduce

Edit crashTest.pl (Attached in zip)
Go to commented line indicating where to insert blanks lines or code.

Add blanks lines. At around 19 blanks lines Komodo crashes.

Or

Add lines of code. At around 5 Komodo crashes.

### Expected results

Komodo stops responding. Need to kill the process.

### Actual results

Komodo stops responding. Need to kill the process.

### Platform Information

Komodo IDE, version 10.2.1, build 89853, platform win32-x86.
Built on Wed Mar 08 12:38:48 2017.

### Additional Information

 
[crashTest.zip](https://github.com/Komodo/KomodoEdit/files/882463/crashTest.zip)

[pystderr.zip](https://github.com/Komodo/KomodoEdit/files/882464/pystderr.zip)

**Comment by mitchell-as** [comment]
I'm not able to reproduce this on Linux  at least, and the error log unfortunately doesn't contain anything interesting. Can you reproduce this problem in safe mode? (Help > Troubleshooting > Restart in Safe Mode)

**Comment by pwo311** [comment]
I have reproduced this in safe mode.
I have reproduced it after whacking the "Komodo IDE" and "KomodoIDE" directory under AppData\Local\Active in normal and safe mode.

**Comment by pwo311** [comment]
[pystdout.zip](https://github.com/Komodo/KomodoEdit/files/882701/pystdout.zip)
[pystderr.zip](https://github.com/Komodo/KomodoEdit/files/882700/pystderr.zip)

Safe mode logs

**Comment by mitchell-as** [comment]
Thanks for being so thorough. Perhaps this is a windows-only issue. @cgchoffman can you reproduce this on Windows?

**Comment by pwo311** [comment]
I actually have gone as far as uninstalling PDK, Komodo and Perl 5.22, whacking all the directories and environment path stuff and then installing everything again. Same problem.

**Comment by pwo311** [comment]
I should add the all the scripts I have had problems with in Komodo execute correctly and produce the expected results outside of Komodo. I checked each script for anything other than ANSI characters and have gone so far as to verify Windows 1252 encoding.

**Comment by Naatan** [comment]
@pwo311 could you please use the crash reporter to send in the crash report and reference this github issue in the comments for the crash report? Thanks!

**Comment by pwo311** [comment]
crash reporter? I am not familiar with this facility. Could you point me in the right direction please?

**Comment by Naatan** [comment]
It should pop up when Komodo crashes.

Perhaps you could clarify what happens when Komodo "crashes" for you?

**Comment by cgchoffman** [comment]
I can't reproduce this on Windows.

**Comment by Naatan** [comment]
Oh sorry, your Komodo is hanging, not crashing. 

I'm unable to repro either. But given you say that it happens even when just writing regular code and on any file I'd wager this is not a widespread issue.

Is this your first time trying Komodo? If not, I assume this didn't happen before? Any idea when it started happening?

**Comment by pwo311** [comment]
I am a long time user. Sometime over the last couple of months Komodo crashed a couple times but I assumed maybe it was OS decay from all the different development environments. I imaged the PC about a month ago and started doing some Perl development a couple of weeks ago and the crashes started happening on a scripts larger than 30-50 lines. Only the one version of Perl installed with Komodo and PDK as well as Python. I started removing variables like Python, etc to see if anything improved. I have even gone so far as to remove all the active state products and references in the path and then installed again. Crashes continue to happen. I did manage to grab some of the Windows temp metadata from the Komodo process termination if you think that might be helpful.

**Comment by Naatan** [comment]
More information definitely doesn't hurt.

Could you maybe try installing an older version of Komodo? 10.1, 10.0, maybe even 9.3.

http://downloads.activestate.com/Komodo/releases/

**Comment by pwo311** [comment]
So 9.3.2 fails as spectacularly as 10.2. Any ideas?

**Comment by Naatan** [comment]
Seeing as you are a long term user and are only now facing this issue I would have to assume that this issue was not introduced by Komodo itself (whether or not Komodo is still the cause is to be determined). Can you think of anything else that might have caused this problem to start happening? Any software installed that might be interfering (virus scanners are the usual culprit).

**Comment by pwo311** [comment]
I agree that it could be something external to Komodo. I have a laptop which is installed pretty close to the same way as they are corporate images. I have no problems on the laptop. I think at this point I will compare DLL versions loaded in the Komodo process space between the two and hope to find a difference.

**Comment by pwo311** [comment]
I believe I have found it and it may not be Komodo. I will have to dig deeper into the API usages in the DLL's.
ActualTools Windows Manager seems to cause Komodo to hurl. I have been able to consistently make Komodo hurl by starting the tool and no problems occur when it is not loaded. I will try running both and excluding the Komodo process from management by the tool to see if Komodo survives the test. If so, at least I have found the culprit and will let ActualTools know. I will post my results later in case anyone else encounters the same problem.
